### ISSUE TITLE
INTERNALERROR> AssertionError: Test coverage should never be `None`

### ISSUE THREAD

**Issue by kierun** [pytest-dev/pytest-cov#124]
When running tests in parallel |(for `-n` of one or more), I get this weird `internalerror` message. However, if I run the tests without the `-n 1` option, everything is fine.

```
; py.test --cov=wireless_id_label_editor -n 1 tests/
============================================================================== test session starts ==============================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/usr/repos/wireless_id_label_editor, inifile: setup.cfg
plugins: pylama-7.0.9, bdd-2.17.0, catchlog-1.2.2, colordots-0.1, cov-2.3.0, html-1.9.0, pep8-1.0.6, xdist-1.14
gw0 [123]
scheduling tests via LoadScheduling
...........................................................................................................................INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/home/usr/.virtualenvs/widle/lib/python2.7/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
Coverage.py warning: No data was collected.


---------- coverage: platform linux2, python 2.7.5-final-0 -----------
Name                                              Stmts   Miss Branch BrPart    Cover   Missing
-----------------------------------------------------------------------------------------------
wireless_id_label_editor/RFC2217_client.py           87      0     12      0   100.0%
wireless_id_label_editor/from_reader.py              84      0     22      0   100.0%
wireless_id_label_editor/function_dispatcher.py      22      0      4      0   100.0%
wireless_id_label_editor/gui.py                     276      0     58      0   100.0%
wireless_id_label_editor/nifty_logger.py             56      0      2      0   100.0%
wireless_id_label_editor/reader.py                   13      0      0      0   100.0%
wireless_id_label_editor/to_reader.py                53      0      0      0   100.0%
-----------------------------------------------------------------------------------------------
TOTAL                                               591      0     98      0   100.0%

```

**Comment by mozbhearsum** [comment]
I can confirm this. A full log from my CI job is at https://public-artifacts.taskcluster.net/N7uOuz-wS4Kaoq0vZpYNVA/0/public/logs/live_backing.log.

**Comment by gst** [comment]
fyi : I have the same assertion error also using with `-n` argument and without then the internalerror disappear

**Comment by foobarna** [comment]
Same thing happens locally with `pytest-cov==2.3.0`.
I've "fixed" it by pinning in tox `pytest-cov==2.2.1`

`INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
FINTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
.INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 94, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/main.py", line 125, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
INTERNALERROR>     wrap_controller.send(call_outcome)
INTERNALERROR>   File "/Users/blink/workspace/flowy/.tox/py35/lib/python3.5/site-packages/pytest_cov/plugin.py", line 221, in pytest_runtestloop
INTERNALERROR>     assert self.cov_total is not None, 'Test coverage should never be `None`'
INTERNALERROR> AssertionError: Test coverage should never be `None`
Coverage.py warning: No data was collected.`

**Comment by ionelmc** [comment]
It would appear that `pytest_runtestloop` hook is being called in the dist slaves (which don't have any summary info). This is a regression caused by https://github.com/pytest-dev/pytest-cov/pull/116 

Hmmm

**Comment by ionelmc** [comment]
fyi, released 2.3.1 with fix
